### PR TITLE
feat(compiler): Moving to tslint parser

### PR DESCRIPTION
**Pull request by nchanged**

### CODE REVIEW

File: src/compiler/Visitor/scopeTracker.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export function scopeTracker(visitor: IVisit): IASTScope {
             }
           } else if (id.type === 'ObjectPattern' && id.properties) {
             for (const prop of id.properties) {
-              if (prop.type === 'Property' && prop.key == prop.value) {
+              if (prop.type === 'Property' && prop.key.name == prop.value.name) {
```
Comment (FKobus) [inline src/compiler/Visitor/scopeTracker.ts]: `===`

File: src/compiler/Visitor/scopeTracker.ts [hunk 2]
```diff
@@ -82,18 +82,20 @@ export function scopeTracker(visitor: IVisit): IASTScope {
     if (node.params) {
       for (const item of node.params) {
         if (item.type === 'Identifier') {
```
Comment (FKobus) [inline src/compiler/Visitor/scopeTracker.ts]: `if (item.type === 'Identifier' && node.body) {` saves a if block ;)

File: src/compiler/transformers/ts/CommonTSfeaturesTransformer.ts [hunk 3]
```diff
@@ -34,29 +35,33 @@ export function CommonTSfeaturesTransformer(): ITransformer {
             return { removeNode: true, ignoreChildren: true };
           }
 
-          if (node.type === 'TypeAssertion' || node.type === 'NonNullExpression') {
+          if (node.type === ASTType.TypeAssertion || node.type === ASTType.NonNullExpression) {
             return { replaceWith: node.expression };
           }
-
-          if (node.type === 'MethodDefinition' && node.value && node.value.type === 'EmptyBodyFunctionExpression') {
+          // EmptyBodyFunctionExpression is a specific buntis key
```
Comment (FKobus) [inline src/compiler/transformers/ts/CommonTSfeaturesTransformer.ts]: is in buntis the body also empty?

if so...
```
if (
  node.type === 'MethodDefinition' &&
  node.value &&
  !node.value.body
) {
```

File: src/compiler/transpilers/testTranspiler.ts [hunk 4]
```diff
@@ -9,6 +9,7 @@ import { makeFuseBoxPath } from '../../utils/utils';
 import { generate } from '../generator/generator';
```
Comment (FKobus) [inline src/compiler/transpilers/testTranspiler.ts]: remove the `import * from buntis`?

File: src/core/Module.ts [hunk 5]
```diff
@@ -142,12 +143,7 @@ export class Module {
     try {
       const withSourcemaps = this.isSourceMapRequired();
       if (!this.isJavascriptModule()) {
-        this.ast = buntis.parseTSModule(this.contents, {
-          directives: true,
-          jsx: this.props.extension === '.tsx',
-          next: true,
-          loc: withSourcemaps,
-        }) as ASTNode;
+        this.ast = parseTypeScript(this.contents, { jsx: this.props.extension === '.tsx', locations: withSourcemaps });
```
Comment (FKobus) [inline src/core/Module.ts]: what happens when people type jsx in ts files?

Comment (nchanged) [reply src/core/Module.ts]: it's not supported by typescript ;-)

File: src/compiler/transformers/ts/CommonTSfeaturesTransformer.ts (hunk 3 above)
Comment (nchanged) [reply src/compiler/transformers/ts/CommonTSfeaturesTransformer.ts]: let's be explicit here ;-)

**Pull request merged**